tests/BreakingChangesComparerTest.php

```diff
@@ -0,0 +1,1283 @@
+<?php
+
+namespace SilverStripe\DeprecationChecker\Tests;
+
+use Doctum\Project;
+use PHPUnit\Framework\TestCase;
+use SilverStripe\DeprecationChecker\Compare\BreakingChangesComparer;
+use SilverStripe\DeprecationChecker\Parse\ParserFactory;
+use Symfony\Component\Console\Output\NullOutput;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Filesystem\Path;
+
+class BreakingChangesComparerTest extends TestCase
+{
+    private static Project $project;
+
+    public static function setUpBeforeClass(): void
+    {
+        // Note that from-only wouldn't be included in the supported modules list
+        // because we only care about modules in the "to" list. See GenerateCommand::findSupportedModules().
+        // But we do have some code in some-org/from-only to check that this list is respected.
+        $supportedModules = [
+            'some-org/module1' => [
+                'type' => 'module',
+                'packagist' => 'some-org/module1',
+            ],
+            'some-org/module2' => [
+                'type' => 'module',
+                'packagist' => 'some-org/module2',
+            ],
+            'some-org/to-only' => [
+                'type' => 'module',
+                'packagist' => 'some-org/to-only',
+            ],
+            'some-org/some-recipe' => [
+                'type' => 'recipe',
+                'packagist' => 'some-org/some-recipe',
+            ],
+            'some-org/some-theme' => [
+                'type' => 'theme',
+                'packagist' => 'some-org/some-theme',
+            ],
+            // We need to include framework with some stubs for some of the type checks
+            // e.g. checking for the configurable trait on DataObject subclasses.
+            'silverstripe/framework' => [
+                'type' => 'module',
+                'packagist' => 'silverstripe/framework',
+            ],
+        ];
+        $factory = new ParserFactory($supportedModules, Path::join(__DIR__, 'fixture-code'));
+        BreakingChangesComparerTest::$project = $factory->buildProject();
+        BreakingChangesComparerTest::$project->parse();
+    }
+
+    public static function tearDownAfterClass(): void
+    {
+        // Remove any cache from running the test.
+        $cacheDir = Path::join(__DIR__, 'fixture-code/cache');
+        if (is_dir($cacheDir)) {
+            $filesystem = new Filesystem();
+            $filesystem->remove($cacheDir);
+        }
+    }
+
+    public function testCompareActions()
+    {
+        $comparer = new BreakingChangesComparer(new NullOutput());
+
+        // There should be no actions before we've started comparing
+        $this->assertEmpty($comparer->getActionsToTake());
+
+        $comparer->compare(BreakingChangesComparerTest::$project);
+        $this->assertEquals($this->getExpectedActions(), $comparer->getActionsToTake(), 'should get expected actions');
+    }
+
+    public function testCompareChanges()
```

Comment:
I got a unit test failure locally for this

Reply:
Please see the note in the PR description about that.